lib/matplotlib/__init__.py

```diff
@@ -1588,214 +1587,107 @@ def foo(ax, *args, **kwargs)
 
     Parameters
     ----------
-    replace_names : list of strings, optional, default: None
+    replace_names : List[str] or None, optional, default: None
         The list of parameter names which arguments should be replaced by
-        `data[name]`. If None, all arguments are replaced if they are
-        included in `data`.
-    replace_all_args : bool, default: False
-        If True, all arguments in *args get replaced, even if they are not
-        in replace_names.
+        ``data[name]`` (if ``data[name]`` does not throw an exception).  If
+        None, replacement is attempted for all arguments.
     label_namer : string, optional, default: None
         The name of the parameter which argument should be used as label, if
         label is not set. If None, the label keyword argument is not set.
-    positional_parameter_names : list of strings or callable, optional
-        The full list of positional parameter names (excluding an explicit
-        `ax`/'self' argument at the first place and including all possible
-        positional parameter in `*args`), in the right order. Can also include
-        all other keyword parameter. Only needed if the wrapped function does
-        contain `*args` and (replace_names is not None or replace_all_args is
-        False). If it is a callable, it will be called with the actual
-        tuple of *args and the data and should return a list like
-        above.
-        NOTE: callables should only be used when the names and order of *args
-        can only be determined at runtime. Please use list of names
-        when the order and names of *args is clear before runtime!
 
     .. note:: decorator also converts MappingView input data to list.
     """
+
+    if func is None:
```

Comment:
This could use an explanatory comment...

Reply:
Done, see also discussion starting at https://github.com/matplotlib/matplotlib/pull/11398#discussion_r197648946.